recipe/meta.yaml

```
@@ -11,7 +14,8 @@ source:
   sha256: fdccab16133c9d927a183c2648bcea8d956fb41eb1df2aacaa73eb0b95e43724
 
 build:
-  number: 0
+  number: 1
+  string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}{{ build_ext }}
```

If I understood #97 correctly, the user wanted to have the non-jemalloc variant. So you might want to make that easier on them by also giving a build string to match for that case. For example
```suggestion
  number: 1
  string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_jemalloc  # [rocksdb_build_ext == "jemalloc"]
  string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_default   # [rocksdb_build_ext != "jemalloc"]
```
Or you could rename `rocksdb_build_ext` to be one of `default` and `jemalloc` and then use it directly:
```suggestion
  number: 1
  string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ rocksdb_build_ext }}
```

---

I prefer what you propose (it makes it easier for users to enforce the `default` builds).

I was searching whether conda-forge had some convention regarding build variants, but I have not found any yet beyond folly's.

---

Side-note: must `track_features` be used for cases like this where a specific allocator could be used by all or parts of an environment's packages?

---

track_features is useful if you want to prioritize one build over the other (the one with `track_features` will need to be explicitly demanded by a user, as the solver won't choose it unless forced to). It's not a must, but perhaps not a bad choice here.

---

One thing to note is that you're making the current behaviour (with jemalloc) - at least naming-wise - the non-default case. Perhaps the default should be with jemalloc, and the opt-in is for avoiding it? 🤔 

Don't know this package well-enough to give an informed opinion about this though.

---

> track_features is useful if you want to prioritize one build over the other (the one with `track_features` will need to be explicitly demanded by a user, as the solver won't choose it unless forced to). It's not a must, but perhaps not a bad choice here, to get consistent behaviour when installing `rocksdb`

OK, I feel it could be used for custom allocators' cases, but this is not required for this PR and beyond the scope of this PR.

> One thing to note is that you're making the current behaviour (with jemalloc) - at least naming-wise - the non-default case. Perhaps the default should be with jemalloc, and the opt-in is for avoiding it? 🤔

> Don't know this package well-enough to give an informed opinion about this though.

[`jemalloc` is an optional dependencies of `rocksdb`](https://github.com/facebook/rocksdb/blob/44b741e9ccc2a99334806c08e8b09c3ad8010f63/CMakeLists.txt#L90), and its usage must be left as the discretion of the end-users who wants to have entire control over memory allocation as custom allocators generally overwrite `malloc` and similar API and (dynamically) override them at runtime.

As specified by @grapland0 (whose analysis I agree with) in #97:

> Currently this library is built with jemalloc enabled, so user binaries linked to this are getting jemalloc enabled and have it handle the memory management everywhere. This might be unexpected and may conflict to some use scenarios, e.g. user wants to have another malloc implementation like tcmalloc, or wants to write an extension for cpython.

Hence, I think it is better not to make the `default` builds use `jemalloc`, be it for `rocksdb` or any other packages.

@grapland0: As a user of `jemalloc`, what do you think of this approach?

---

Looks good to me if we can get a rocksdb lib with no jemalloc by plain `conda install rocksdb`, where I guess the "default" works.

There are also some package maintainers from Arch and Frdora tends to make the build jemalloc-free: https://gitlab.archlinux.org/archlinux/packaging/packages/rocksdb/-/issues/2 . So it is a common practise.

---

The `default` build should be preferred over the `jemalloc` because it has fewer dependencies (this is something packages managers take into account).